src/raftstore/store/peer_storage.rs

```diff
@@ -103,6 +104,8 @@ pub fn first_index(state: &RaftApplyState) -> u64 {
     state.get_truncated_state().get_index() + 1
 }
 
+const DEFAULT_CACHE_SIZE: usize = 3000;
```

Comment:
If we have many regions like > 10000 in one store, and in most cases, there are only some active regions, so use 3000 for all regions seem wasteful. 

Maybe we can create the RingQueue with a small capacity and a large max size.